sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -164,9 +166,17 @@ export class BatchingReceiver extends MessageReceiver {
           this._receiver.drain = true;
           this._receiver.addCredit(1);
 
-          const drainTimeout = delay(receiveDrainTimeoutInMs);
+          const drainTimeout = delay(this._drainTimeoutInMs).then(() => {
+            return Promise.reject();
+          });
+
           // Wait for the drain event to be fired, or for the timeout.
-          await Promise.race([drainPromise, drainTimeout]);
+          try {
+            await Promise.race([drainPromise, drainTimeout]);
+          } catch {
+            // Close the receiver link since we have not received the receiver drain event.
+            await this.onDetached();
```

Comment:
Why call `onDetached()` instead of `close()`?

Reply:
I updated to use `close()` instead. I thought since `close` sets `wasCloseInitiated` it might impact recovering from network issues, but that's not actually an issue since `drain` should be called only when `receiveMessages` is going to actually return/throw, and the next `receiveMessages` call will just create a new `batchingReceiver` anyway.